### DISCUSSION TITLE
KubeVela有官方的GoLang SDK吗？

### DISCUSSION THREAD

**Discussion by kissycn** [kubevela/kubevela#6124] [Q&A]
我需要通过一些KubeVela的API来实现自己一些业务逻辑处理。
我们系统是通过Go语言开发的，在官网文档中找到了[RestAPI文档](https://kubevela.io/zh/docs/platform-engineers/openapi/overview)
为了简化调用逻辑我想使用GoSDK，于是找到了 [kubevela-go-sdk](https://github.com/kubevela-contrib/kubevela-go-sdk)
看了文档比较简单所用有几个问题想问下：
1、kubevela-go-sdk 是不是官方的sdk
2、是否与RestAPI版本保持一致
3、我在kubevela-go-sdk 中没有找到 [对多集群的那部分SDK ](https://kubevela.stoplight.io/docs/kubevela/5e68399450a00-list-all-clusters)  是没有吗

**Comment by Somefive** [comment]
1. 基本算是官方SDK
2. RestAPI 是面向 VelaUX，是 KubeVela Application API 在业务层的封装。kubevela-go-sdk, vela CLI, kubectl 这些都是面向原生 Kubernetes 生态的，不完全一致，可以按需使用，不冲突。
3. 多集群的部分如果你使用 SDK 的话可以直接配置 topology policy。

**Comment by kissycn** [comment]
@Somefive  首先感谢您的回复。
多集群的API我想使用的是，clister：list、create、status等，go-sdk是不是没有支持这类API，我看RestAPI是有的。

**Reply by Somefive** [reply]
嗯，如果你说的不是应用级别的，而是系统管控级别的，那么目前确实没有这样的 go-sdk。https://github.com/oam-dev/cluster-gateway/pull/139 目前正在做一些相关的代码重构，估计需要一部分时间。

**Comment by chivalryq** [comment]
关于2. RestAPI 我补充一些：RestAPI 还应该有一层SDK，负责把Rest接口转换为编程语言的方法（或函数）。目前这个项目正在由社区贡献：https://github.com/kubevela/kubevela/issues/5428。
相比之下，kubevela-go-sdk 的作用是面向集群内部的 Application 资源级别的。
